### DISCUSSION TITLE
Detecting and showing Keyboard status

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#14160] [Q&A]
Writing frontends isn't my thing but I've been assigned to it so ...

The frontend has a line of status boxes at the bottom of the front window. Here's part of it
```xaml
            <TextBox IsReadOnly="True" Name="CAPS" Height="30" Width="100" VerticalAlignment="Center" HorizontalAlignment="Center" HorizontalContentAlignment="Center"></TextBox>
            <TextBox IsReadOnly="True" Name="INS"  Height="30" Width="100" VerticalAlignment="Center" HorizontalAlignment="Center" HorizontalContentAlignment="Center"></TextBox>
            <TextBox IsReadOnly="True" Name="NUM"  Height="30" Width="100" VerticalAlignment="Center" HorizontalAlignment="Center" HorizontalContentAlignment="Center"></TextBox>
            <TextBox IsReadOnly="True" Name="SCRL" Height="30" Width="100" VerticalAlignment="Center" HorizontalAlignment="Center" HorizontalContentAlignment="Center"></TextBox>
        </StackPanel>
```
How does one fill these boxes? The contractor who wrote the original version (without whom I would not have known where even to begin) had keyboard detection happening in two places. (I've since made a mess of them)
```cs
        private void RichEditBox_KeyDown(object sender, KeyRoutedEventArgs e)
        {
            if (e.Key == VirtualKey.CapitalLock)
            {
                CAPS.Text = Console.CapsLock ? "CAPS" : "";
            }
```
and 
```cs
        private async void MainWindow_Loaded(object sender, RoutedEventArgs e)
        {
            CAPS.Text = Console.CapsLock ? "CAPS" : "";
```
That's the CAPS handling. How do I do NumLock, and InsertMode and ScrollLock?

On a related note, I'd prefer to be able to have "CAPS" greyed-out when off and in bold when on. I expect that `TextBox` probably isn't the best choice for that. Suggestions?

-Bruce

**Comment by jeromelaban** [comment]
You have access to numlock (`NumberKeyLock`) through the same keydown event, though the behavior may vary between platforms. What's your target platform?

**Reply by axtens** [reply]
Initially, Windows. All others eventually

**Reply by jeromelaban** [reply]
Thanks. So what's the issue then? Windows should be providing you with the key event for numlock.

**Comment by axtens** [comment]
This is Q&A right? So if I don't know how to do something, is this the
wrong place in which to ask it? I've more or less figured out CAPS and NUM
but not INS or SCRL.

**Reply by jeromelaban** [reply]
It's a community Q&A place, though not all answers are available, particularly when dealing with WinAppSDK specifically (we're not managing windows).

When you say INS or SCRL is not figured out, what's happening?
